decoder/pokemon.go

```diff
@@ -740,6 +755,37 @@ func (pokemon *Pokemon) updatePokemonFromDiskEncounterProto(ctx context.Context,
 	updateStats(ctx, db, pokemon.Id, stats_lureEncounter)
 }
 
+func (pokemon *Pokemon) setPokemonDisplay(pokemonId int16, display *pogo.PokemonDisplayProto) {
+	pokemon.Gender = null.IntFrom(int64(display.Gender))
+	pokemon.Form = null.IntFrom(int64(display.Form))
+	pokemon.Costume = null.IntFrom(int64(display.Costume))
+	pokemon.setWeather(null.IntFrom(int64(display.WeatherBoostedCondition)))
+	if pokemon.PokemonId == 0 || !pokemon.IsDitto {
+		pokemon.PokemonId = pokemonId
+	}
+}
+
+func (pokemon *Pokemon) setWeather(weather null.Int) {
+	if pokemon.PokemonId != 0 && pokemon.Weather.ValueOrZero() != weather.ValueOrZero() {
+		if pokemon.IsDitto {
+			if weather.ValueOrZero() == 3 {
+				// both Ditto and disguise are boosted and Ditto was not boosted: none -> boosted
+				//pokemon.Level += null.IntFrom(5)
```

Comment:
is there an easy way to increment the value of a null.Int prop? :)

Reply:
Can you do

pokemonLevel.Value += 5
?
Does that work?

Reply:
that did not work, but I pushed a possibility where the compiler did not complain

Reply:
I misremembered the struct member names. That is what I meant and I think it’s fine as the struct just contains two bits